src/thor/triplegbuilder.cc

```diff
@@ -1560,12 +1569,13 @@ void TripLegBuilder::Build(
       auto edge_length = static_cast<float>(directededge->length());
       trim_shape(edge_begin_info.distance_along * edge_length, edge_begin_info.vertex,
                  edge_end_info.distance_along * edge_length, edge_end_info.vertex, edge_shape);
-      // Add edge shape to trip
-      trip_shape.insert(trip_shape.end(),
-                        (edge_shape.begin() + ((edge_begin_info.trim || is_first_edge) ? 0 : 1)),
+      // Add edge shape to the trip and skip the first point when its redundant with the previous edge
+      // TODO: uncommment correct removal of redundant shape after odin can handle uturns
+      // trip_shape.insert(trip_shape.end(), edge_shape.begin() + !is_first_edge, edge_shape.end());
+      trip_shape.insert(trip_shape.end(), edge_shape.begin() + !edge_begin_info.trim,
```

Comment:
the commented out line above is the correct code and causes no duplicates to be had in the shape. however at uturns, it seems that odin does not detect the uturn has occured unless the shape point is doubled where the uturn occurs. cc @dgearhart